Sources/Sentry/Profiling/SentryProfiledTracerConcurrency.mm

```diff
@@ -48,22 +48,49 @@
 
     if (_gProfilersToTracers == nil) {
         _gProfilersToTracers = [NSMutableDictionary</* SentryProfiler.profileId */ NSString *,
-            NSMutableSet<SentryTracer *> *> dictionaryWithObject:[NSMutableSet setWithObject:tracer]
-                                                          forKey:profilerKey];
+            /* number of in-flight tracers */ NSNumber *>
+            dictionary];
         _gTracersToProfilers =
             [NSMutableDictionary</* SentryTracer.tracerId */ NSString *, SentryProfiler *>
-                dictionaryWithObject:profiler
-                              forKey:tracerKey];
+                dictionary];
+    }
+
+    _gProfilersToTracers[profilerKey] = @(_gProfilersToTracers[profilerKey].unsignedIntValue + 1);
+    _gTracersToProfilers[tracerKey] = profiler;
```

Comment:
I'd prefer to keep the stricter type safety here, since if it's just a SentryId, it could mistakenly get called with a SentryId from something other than a tracer in the future.